pkg/azuredisk/controllerserver.go

```diff
@@ -838,12 +847,19 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 			}
 		case consts.ResourceGroupField:
 			resourceGroup = v
+		case consts.UserAgentField:
+			newUserAgent := v
+			localCloud, err = azureutils.GetCloudProvider(d.kubeconfig, d.cloudConfigSecretName, d.cloudConfigSecretNamespace, newUserAgent)
```

Comment:
Should return error directly, make `CreateSnapshot' fail:
```
return fmt.Errorf("create cloud with UserAgent(%s) failed with: (%s)", newUserAgent, err)
```

Reply:
Shouldn't this be the following?

```go
return status.Errorf(codes.InvalidArgument, "create cloud with UserAgent(%s) failed with: (%s)", newUserAgent, err)
```